### PR TITLE
Utils: Deprecate buildTermTree function

### PR DESCRIPTION
**Pull request by gziolo**
Part of #3955.

This PR duplicates `buildTermsTree` function and puts it closer to the usage. We took this approach to make it possible to get rid of `utils` module at all. In the long term, we might want to seek to remove duplication by moving all the logic that build terms tree closer to the data layer which is where it should really happen.

As part of this PR we completely deprecate (remove) `wp.utils.buildTermsTree`.

### CODE REVIEW

File: editor/utils/test/terms.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Internal dependencies
+ */
+import { buildTermsTree } from '../terms';
```
Comment (youknowriad) [inline editor/utils/test/terms.js]: Do we really need the test twice?

**Review by youknowriad — APPROVED** [review]
I agree with this change, this function is too specific to be really valuable as an exposed API

File: editor/utils/test/terms.js (hunk 1 above)
Comment (gziolo) [reply editor/utils/test/terms.js]: In case it diverges, it might make sense.

**Comment by chrisvanpatten** [comment]
I'm curious why this decision was made?

Right now we offer a public `TreeSelect` component, and the ability to fetch terms (or hierarchical post types) through the data store, but without this — the function that actually bridges the two together — users have to reimplement this logic.

**Pull request merged**